build_tools/circle/build_doc.sh

```diff
@@ -58,32 +58,45 @@ get_build_type() {
         return
     fi
     changed_examples=$(echo "$filenames" | grep -E "^examples/(.*/)*plot_")
-    if [ -n "$filenames" ]
-    then
-        for af in ${filenames[@]}
-        do
-            if [ $af != "build_tools/circle/build_doc.sh" ]
-            then
-                page_figure=$(grep figure $af | grep auto_example | awk -F "/" '{print $NF}' | sed 's/sphx_glr_//' | awk -F "_" '{OFS="_";$NF=""; print $0}')
-                page_image=$(grep image $af | grep auto_example | awk -F "/" '{print $NF}' | sed 's/sphx_glr_//' | awk -F "_" '{OFS="_";$NF=""; print $0}')
-            fi
-        done
-    fi
-    if [ -n "$page_figure" ]
-    then
-        changed_examples+=$(echo ${page_figure::-1} | sed 's/_ /|/g')
-    fi
-    if [ -n "$page_image" ]
+
+    if [[ -n "$filenames" ]]
     then
-        changed_examples+=$(echo ${page_image::-1} | sed 's/_ /|/g')
+        # get rst files
+        rst_files="$(echo "$filenames" | grep -E "rst$")"
+
+        # get lines with figure or images
+        img_fig_lines="$(echo "$rst_files" | xargs grep -shE "(figure|image)::")"
+
+        # get only auto_examples
+        auto_example_files="$(echo "$img_fig_lines" | grep auto_examples | awk -F "/" '{print $NF}')"
+
+        # remove "sphx_glr_" from path and accept replace _\d\d\d.png with .py
+        image_paths="$(echo "$auto_example_files" | sed 's/sphx_glr_//' | sed -e 's/_[[:digit:]][[:digit:]][[:digit:]].png/.py/')"
+
+        # get unique values
+        examples_in_rst="$(echo "$image_paths" | uniq | paste -sd '|')"
```

Comment:
Can we updated changed examples here by:

```bash
if [[ -n "$examples_in_rst" ]] then
	changed_examples="$changed_examples\n$examples_in_rst"
fi
```

The rest of the code from here can be left unchanged.

Reply:
If you take the example of [cmarmo#5](https://github.com/cmarmo/scikit-learn/pull/5), in my local bash, the code
`changed_examples="$changed_examples\n$examples_in_rst"` 
output
`+ changed_examples='examples/svm/plot_custom_kernel.py
examples/svm/plot_iris_svc.py\nplot_cluster_comparison.py
plot_kmeans_assumptions.py
plot_kmeans_digits.py
plot_mini_batch_kmeans.py
plot_affinity_propagation.py
plot_mean_shift.py
plot_segmentation_toy.py
plot_coin_segmentation.py
plot_linkage_comparison.py
plot_agglomerative_dendrogram.py
plot_ward_structured_vs_unstructured.py
plot_agglomerative_clustering.py
plot_agglomerative_clustering_metrics.py
plot_dbscan.py
plot_optics.py
plot_birch_vs_minibatchkmeans.py
plot_adjusted_for_chance_measures.py
plot_tree_regression.py
plot_iris_dtc.py
plot_tree_regression.py
plot_tree_regression_multioutput.py
plot_multioutput_face_completion.py'
`
The `\n` character is printed as is, it is not translated in an `EOL`... if this is different from your bash, this is a portability issue anyway.
I think the actual version is ok from this point of view. WDYT?

Reply:
Is there anyway to add the `$examples_in_rst` into `$changed_examples` so we leave the rest of the code unchanged?

Reply:
That's all my drama... :( .. could not find it.

Reply:
@thomasjpfan you've got it! :)